### PR TITLE
[Artwork/Sidebar] Make sale components fully dynamic

### PR DESCRIPTION
**Pull request by alloy**
Makes use of the following changes to the MP schema: https://github.com/artsy/metaphysics/pull/1146

### CODE REVIEW

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx [hunk 1]
```diff
@@ -10,31 +10,25 @@ import { ArtworkSidebarBidAction_artwork } from "__generated__/ArtworkSidebarBid
...
 }
 
 export class ArtworkSidebarBidAction extends React.Component<
   ArtworkSidebarBidActionProps
 > {
   render() {
-    const { artwork, me } = this.props
-    const registrationAttempted = me && me.bidders && me.bidders.length > 0
+    const { artwork } = this.props
+    const registrationAttempted = !!artwork.sale.registrationStatus
```
Comment (oxaudo) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx]: Is the `!!` means that the attribute is present and not blank? I always get confused with those double negative things...

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx [hunk 2]
```diff
@@ -107,7 +101,15 @@ export const ArtworkSidebarBidActionFragmentContainer = createFragmentContainer(
   ArtworkSidebarBidAction,
   graphql`
     fragment ArtworkSidebarBidAction_artwork on Artwork {
+      bidderStatus(live: true) {
+        active_bid {
+          __id
+        }
+      }
       sale {
+        registrationStatus {
```
Comment (oxaudo) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx]: Really much cleaner to have it under artwork. I was wondering how would we deal with all this user/sale/artwork ids passing around. Thanks for taking it!

File: src/Apps/Artwork/Components/__stories__/ArtworkSidebar/ArtworkSidebarBidActionStory.story.tsx [hunk 3]
```diff
@@ -21,99 +22,137 @@ storiesOf("Styleguide/Artwork/Sidebar", module).add("BidAction", () => {
       {/* Auction in preview state */}
       <Section title="Auction Preview / Bidder not registered">
         <BidAction
-          artwork={ArtworkFromAuctionPreview}
-          me={NotRegisteredToBid}
+          artwork={merge({}, ArtworkFromAuctionPreview, NotRegisteredToBid)}
```
Comment (oxaudo) [inline src/Apps/Artwork/Components/__stories__/ArtworkSidebar/ArtworkSidebarBidActionStory.story.tsx]: 👍

**Comment by oxaudo** [comment]
@alloy - the changes look good to me! I assume the MP change went through already. Not sure what is up with this circle error above. Please merge on green once that is figured out.

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx (hunk 1 above)
Comment (alloy) [reply src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx]: Use a REPL when in doubt!

```
$ node
> !!{}
true
> !!null
false
```

So a double bang converts truthy/falsy values into absolute booleans.

**Comment by alloy** [comment]
Fixed linter issue, merging 👍

**Pull request merged**